test_schedule.py

```diff
@@ -258,3 +258,20 @@ def stop_job():
 
         schedule.cancel_job(mj)
         assert len(schedule.jobs) == 0
+
+    def test_clear_by_tag(self):
```

Comment:
Cool, can we add some cases here to:
* ensure invalid tag types (e.g non str/int) raise an exception
* ensure `.tag('tag1', 'tag1', 'tag1')` (reusing the same tag) doesn't cause havoc